Controllers/PostController.cs

```diff
@@ -27,7 +28,59 @@ public async Task<ActionResult> CreatePost([FromBody] PostDto post)
                 var result = await postService.Create(post.GetPersistentObject());
                 return Created(result.Id.ToString(), result);
             }
-            return BadRequest(post.Errors);
+            return BadRequest(post.Error);
+        }
+
+        [HttpGet("{id}")]
+        public async Task<ActionResult<Post>> GetPost(Guid id)
+        {
+            var result = await postService.Get(id);
+            if(result == null)
+            {
+                return NotFound();
+            }
+            return result;
+        }
+
+        [HttpGet("page/{number}")]
+        public async Task<ActionResult<List<Post>>> GetPaginate(int number)
+        {
+            var queryOrder = Request.Query["order"].ToString();
+            if(!String.IsNullOrEmpty(queryOrder))
+            {
+                queryOrder = queryOrder.ToLower();
+                if(queryOrder == "desc")
+                {
+                    return await postService.GetPage(number, 10 ,true);
+                }
+            }
+            return await postService.GetPage(number);
+        }
+
+        [HttpGet]
+        public async Task<ActionResult<List<Post>>> GetAllPosts()
+        {
+            return await postService.GetAll();
+        }
+
+        [HttpPut]
+        public async Task<ActionResult<Post>> UpdatePost(PostDto post)
+        {
+            if(post.Id == Guid.Empty)
+            {
+                return BadRequest();
+            }
+            if(!post.IsValid())
```

Comment:
Two bad request return point? This is one kind of code smell.
Can we somehow validate within the IsValid method as discussed?

Reply:
will be update on next phase